### PR TITLE
Emailer Update and Fixes

### LINKED ISSUE #911: Fix Emailer
This is broken on our internal vc.

### PR DESCRIPTION
**Pull request by craig8**
This patch updates the emailer and fixes #911 

Included is a test send email script that when ran will connect and send an email via the alert system as well as through the pubsub system.

Note you must configure the emaileragent correctly for this to happen and have a valid smtp address.

---
This change is [Reviewable](https://reviewable.io/reviews/volttron/volttron/912)

### CODE REVIEW

File: services/core/EmailerAgent/emailer/agent.py [hunk 1]
```diff
@@ -54,166 +54,315 @@
...
-    @PubSub.subscribe('pubsub',
-                      prefix=topics.ALERTS.format(agent_class='',
-                      agent_uuid=''))
-    def onmessage(self, peer, sender, bus, topic, headers, message):
-        _log.info('Alert message for topic: {}'.format(topic))
-        _log.debug("Sending mail for message: {}".format(message))
+            send_successful = True
+        except Exception as e:
+            _log.error(
+                'Unable to send email message: %s' % mime_message.as_string())
```
Comment (jhaack) [inline services/core/EmailerAgent/emailer/agent.py]: Update status for failure to send email so it is visible in the VC UI. Obviously, sending an alert would be pointless if you can't send an email.

File: services/core/EmailerAgent/emailer/agent.py [hunk 2]
```diff
@@ -54,166 +54,315 @@
...
+        """
+        RPC Method allowing a platform to send an email address.
+
+        One can also send an email through the pubsub mechanism.
+
+        :param from_address:
+        :param to_addresses:
+        :param subject:
+        :param message:
+        :return:
```
Comment (jhaack) [inline services/core/EmailerAgent/emailer/agent.py]: Remove

File: services/core/EmailerAgent/emailer/agent.py [hunk 3]
```diff
@@ -54,166 +54,315 @@
...
+        """
+        Callback for alert messages that come into the platform.
+
+        :param peer:
+        :param sender:
+        :param bus:
+        :param topic:
+        :param headers:
+        :param message:
+        :return:
```
Comment (jhaack) [inline services/core/EmailerAgent/emailer/agent.py]: Remove

File: services/core/EmailerAgent/emailer/agent.py (hunk 1 above)
Comment (craig8) [inline services/core/EmailerAgent/emailer/agent.py]: Added in addition a success status change if the send was successful.

**Pull request merged**